src/lib.rs

```diff
@@ -26,7 +26,19 @@ mod rand_functions {
         rand::random()
     }
 
-    #[cfg(feature = "shuffle")]
+    #[cfg(feature = "array_functions")]
+    #[rhai_fn(global)]
+    pub fn sample(array: &mut Array) -> rhai::Dynamic {
```

Comment:
You should also provide an additional version of `sample` taking a single `INT` parameter, which samples a number of random elements, returning an array...

You may need to decide whether to allow duplicates...

If allowing duplicates, just loop the right number of times.

If not allowing duplicates, then you need to run through each item checking a random number based on `N / M` probability, where `N` is the number of items to sample and `M` is the total number of items in the array.  Of course, if `N >= M` then the entire array can simply be returned.

Reply:
Great idea. I think the expected behavior of a `.sample(5)` would be to sample 5 unique elements, otherwise the user could call `my_array.sample()` 5 times. I have addressed the other comments but will push (likely tomorrow) once I have this one implemented.